src/api/entities/Portfolio/types.ts

```diff
@@ -1,9 +1,25 @@
 import BigNumber from 'bignumber.js';
 
-import { SecurityToken } from '~/internal';
+import { Leg } from '~/api/entities/Instruction/types';
+import { Account, SecurityToken } from '~/internal';
+import {
+  SettlementDirectionEnum as SettlementDirection,
+  SettlementResultEnum as SettlementResult,
+} from '~/middleware/types';
 
 export interface PortfolioBalance {
   token: SecurityToken;
   total: BigNumber;
   locked: BigNumber;
 }
+
+export interface SettlementLeg extends Leg {
+  direction: SettlementDirection;
+}
+
+export interface HistoricSettlement {
+  blockNumber: BigNumber;
+  status: SettlementResult;
+  accounts: Account[];
```

Comment:
Add a JSDOC comment above this property that says "array of accounts that participated by affirming the settlement"